### PR TITLE
7pm b /mapache slack bot for Google Search

### PR DESCRIPTION
**Pull request by pconrad**
In this PR, we've added the search /mapache search googlecommand. I judge this to be 4 points in that it was quite a bit more difficult in implementing than /mapache status in that I had to learn how to learn how to use the jsoup library to parse html and then learn how to import jsoup through the pom.xml file.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,6 +27,14 @@
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
 
+		<!--jsoup added for search google function-->
```
Comment (pconrad) [inline pom.xml:30]: This can probably be deleted at this point.. no longer needed

File: pom.xml [hunk 2]
```diff
@@ -123,6 +131,28 @@
...
+    		</dependency>
+
+		<!-- https://mvnrepository.com/artifact/net.codebox/javabean-tester -->
+		<dependency>
+			<groupId>net.codebox</groupId>
+			<artifactId>javabean-tester</artifactId>
+			<version>1.0.0</version>
+		</dependency>
+
+		<!-- https://mvnrepository.com/artifact/me.ramswaroop.jbot/jbot -->
```
Comment (pconrad) [inline pom.xml:149]: This dependency (`me.ramswaroop.jbot`) is listed twice... we can delete the second one.

File: src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java [hunk 3]
```diff
@@ -18,6 +21,25 @@
 import edu.ucsb.mapache.models.SlackSlashCommandParams;
 import edu.ucsb.mapache.repositories.ChannelRepository;
 
+import edu.ucsb.mapache.services.GoogleSearchService;
+
+// imports for google search
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java:30]: Can delete these imports probably

File: src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java [hunk 4]
```diff
@@ -18,6 +21,25 @@
...
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import java.util.HashMap;
+
+import edu.ucsb.mapache.models.SearchParameters;
+//
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java:40]: delete this line?

File: src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java [hunk 5]
```diff
@@ -33,13 +55,21 @@
...
     ChannelRepository channelRepository;
 
+    @Autowired
+    GoogleSearchService googleSearchService;
+
+
     /**
      * The token you get while creating a new Slash Command. You should paste the
      * token in application.properties file.
      */
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java:65]: delete this comment..  it's false.  the token goes in secrets-localhost.properties and secrets-heroku.properties

File: src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java [hunk 6]
```diff
@@ -107,6 +136,14 @@ public RichMessage onReceiveSlashCommand(@RequestParam("token") String token,
             return debugCommand(params);
         }
 
+
+
+        ///////
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java:141]: Delete ///// comment?

**Review by kokinarimoto — APPROVED** [review]
Looks good to me!

File: pom.xml [hunk 7]
```diff
@@ -27,6 +27,13 @@
 			<artifactId>spring-boot-starter-web</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.jsoup</groupId>
+			<artifactId>jsoup</artifactId>
+			<version>1.13.1</version>
+		</dependency>
```
Comment (btk5h) [inline pom.xml:34]: Is this dependency necessary anymore?

File: src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java [hunk 8]
```diff
@@ -86,8 +102,7 @@ public RichMessage onReceiveSlashCommand(@RequestParam("token") String token,
         params.setText(text);
         params.setResponseUrl(responseUrl);
 
-        
-        String[] textParts = params.getTextParts();
+        String[] textParts = params.getTextParts();//
```
Comment (btk5h) [inline src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java:105]: Trailing `//` can be removed

File: pom.xml (hunk 7 above)
Comment (XuperBryan) [reply pom.xml]: Thanks for the suggestion, removed the code now

File: src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java (hunk 8 above)
Comment (XuperBryan) [reply src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java]: Thanks, fixed now!

**Review by XuperBryan — APPROVED** [review]
Code looks good to me now

File: pom.xml [hunk 9]
```diff
@@ -124,6 +123,14 @@
 			<artifactId>javabean-tester</artifactId>
 			<version>1.0.0</version>
 		</dependency>
+
+		<!-- https://mvnrepository.com/artifact/me.ramswaroop.jbot/jbot -->
+		<dependency>
```
Comment (mara-downing) [inline pom.xml:128]: I believe this is still an extra dependency since you have the 4.1.0 also listed

**Review by mara-downing — CHANGES_REQUESTED** [review]
Everything looks good except the extra dependency, if you take that out then I think this PR will be ready to merge!

File: pom.xml (hunk 9 above)
Comment (XuperBryan) [reply pom.xml]: fixed now, thanks!

**Review by XuperBryan — APPROVED** [review]
Made all changes suggested by staff, should be good to go!

**Review by mara-downing — APPROVED** [review]
LGTM

**Pull request merged**